skunk/rust/skunk/src/ir_gen/codegen/examples_codegen.rs

```diff
@@ -263,16 +263,20 @@ pub fn example_check_codegen<'ctx>(cg: &mut CodegenState<'ctx>, module: &'ctx as
   Ok(function)
 }
 
+// A main function for examples. This runs all examples in all provided modules. Note that it doesn't take
+// a CodegenState because generating this isn't part of the main codegen flow - instead, it's an optional
+// extra triggered from main.rs.
 pub fn main_for_examples<'ctx>(context: &'ctx Context, target_machine: &TargetMachine, target_triple: &TargetTriple, modules: &Vec<Module<'ctx>>) -> CodegenResult<Module<'ctx>> {
 
-  let cg = CodegenState::new(context, target_machine, target_triple, "main");
+  let mut cg = CodegenState::new(context, target_machine, target_triple, "main");
   let function = cg.module.add_function("main", context.i32_type().fn_type(&[], false), None);
 
   let entry = cg.context.append_basic_block(function, "entry");
   cg.builder.position_at_end(entry);
 
   for submodule in modules {
-    let fn_name = submodule.get_name().to_str().unwrap().to_string() + "_run_examples";
+    let name = submodule.get_name().to_str().unwrap().to_string();
+    let fn_name = name.clone() + "_run_examples";
```

Comment:
Nit: I believe using format is considered idiomatic and is supposed to optimize better repeated string additions (but I imagine in practice its actually very similar):
e.g. 
```
let name = submodule.get_name().to_str().unwrap();
let fn_name = format!("{}_run_examples", &name)
```